Description: Changes to enable package build.
 Changes to the upstream Makefile to enable package builds.
Author: Peter Karlsson <peterk@debian.org>
Author: Robert James Clay <jame@rocasa.us>
Origin: vendor
Last-Update: 2015-09-20

--- a/Makefile
+++ b/Makefile
@@ -6,16 +6,25 @@
 
 # General options (passed to mmail/Makefile and interfac/Makefile):
 
-# With debug:
-#OPTS = -g -Wall -Wextra -pedantic -Wno-char-subscripts
+# Edited for Debian - build with debugging symbols; make it possible to
+# build without optimization
+OPTS = -g -Wall -Wextra -pedantic -fstack-protector-strong -Wformat -Werror=format-security  -Wno-char-subscripts
+ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
+OPTS += -O0
+else
+OPTS += -O2
+endif
 
-# Optimized, no debug:
-OPTS = -O2 -Wall -pedantic -Wno-char-subscripts
+# Edited for Debian - Include CPPFLAGS in build.
+OPTS += $(CPPFLAGS)
+
+# Edited for Debian
+DESTDIR =
 
 # PREFIX is the base directory under which to install the binary and man 
 # page; generally either /usr/local or /usr (or perhaps /opt...):
 
-PREFIX = /usr/local
+PREFIX = $(DESTDIR)/usr
 
 # Delete command ("rm" or "del", as appropriate):
 
@@ -63,7 +72,8 @@
 #--------------------------------------------------------------
 #--------------------------------------------------------------
 
-HELPDIR = $(PREFIX)/man/man1
+HELPDIR = $(PREFIX)/share/man/man1
+DOCDIR = $(PREFIX)/share/doc/multimail
 
 all:	mm
 
@@ -77,7 +87,7 @@
 		CURS_DIR="$(CURS_DIR)" intrfc $(SEP) cd ..
 
 mm:	mm-main intrfc
-	$(CXX) -o mm mmail/*.o interfac/*.o -L$(CURS_LIB) $(LIBS)
+	$(CXX) $(OPTS) $(LDFLAGS) -o mm mmail/*.o interfac/*.o -L$(CURS_LIB) $(LIBS)
 	$(POST)
 
 dep:
@@ -93,7 +103,11 @@
 	cd mmail $(SEP) $(MAKE) RM="$(RM)" modclean $(SEP) cd ..
 
 install::
-	install -c -s mm $(PREFIX)/bin
+	install -d $(PREFIX)/bin $(HELPDIR) $(DOCDIR)
+	install -c $(INSTALL_OPTS) mm $(PREFIX)/bin
 	install -c -m 644 mm.1 $(HELPDIR)
 	$(RM) $(HELPDIR)/mmail.1
-	ln $(HELPDIR)/mm.1 $(HELPDIR)/mmail.1
+#	ln $(HELPDIR)/mm.1 $(HELPDIR)/mmail.1
+	install -m 644 README $(DOCDIR)
+	install -d $(DOCDIR)/colors
+	cp -a colors/* $(DOCDIR)/colors
